similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -218,8 +223,7 @@ private boolean filterPatient(Patient matchPatient, Family family, EntityReferen
             return true;
         }
         // filter out patients with access level less than defined access level threshold
-        AccessLevel patientAccessLevel =
-            this.accessHelper.getAccessLevel(matchPatient, this.accessHelper.getCurrentUser());
+        AccessLevel patientAccessLevel = this.permissions.getPatientAccess(matchPatient).getAccessLevel();
```

Comment:
I approve the change but object the commit message.

Commit message says "I switched from using class OLD to using class NEW, because NEW checks X and OLD does not check X but checks Y instead".

From that a code reviewer can derive that before we only checked for Y but now we check for X instead. In fact, NEW checks for X and then internally calls OLD to check for Y, so now we check for both X and Y, which is totally not clear form the commit message without digging deep into the code

Reply:
The commit message does not have the word `instead`. It clearly says that methods `return different results`. `DefaultPatientAccess` considers owner or Admin cases, while `PatientAccessHelper` does not consider those situations but runs checks of access levels for all groups user belongs to. No other words about what `DefaultPatientAccess` does or does not.

Reply:
you have the words `but runs checks` which normally means "instead". E.g. "I don't know how to play piano, but i know how to play violin". While in the example above, both of them run those checks

Reply:
`but` was entirely in the part about what PatientAccessHelper does, not DefaultPatientAccess. Thats why `but` was not for comparing two methods.
There the sentence was constructed as: "Method X does not do THIS but does THAT." 
PatientAccessHelper `does not consider those situations` but `runs checks of access levels for all groups user belongs to transitionally`. And it literally does not consider owner and Admin, but what it does is runs other checks for groups.

Reply:
Your statement is correct. But put together like that it does imply that one does one thing while the other does the other (while, again, in fact one does both while the other does just one)

IMHO a better way to phrase that would simple be `switched form using OLD to NEW because OLD does not check case X` which explains the problem and the fix in the most concise way. Or with more explanations: `switched form using OLD to NEW because NEW internally calls OLD but also checks X which OLD does not check` or `switched form using OLD to NEW because OLD does not check case X while NEW checks X and otherwise works like OLD` which makes it crystal clear that most cases will stay the same, while cases which match X are handled differently (and only those cases)

Reply:
Well, the commit message says `Used PermissionsManager instead of PatientAccessHelper to compute AccessLevel, because they return different results (...)` and in `()` I gave exact difference on the two methods: considering or not owner and admin cases. Statements are correct. The understanding and further investigation of PT code is up to reader and not relevant to this particular fix.

Reply:
It doesn't imply, because `but` was not between methods comparing them. It was only inside the part about what PatientAccessHelper does.

Reply:
not exactly, the phrase was about both of them: `X does A, Y does not do A but Y does B`. It sure implies "X does A, but not B while Y does B but not A".

If you said "X does A and calls Y, which does not do A but does B" it would be clear who do and doesn't do what

Reply:
Also, I was no explaining how X works. I said "X considers A while Y does not consider A" rest is lyrics :)

Reply:
Our context is not abstract logical statements. Our context is making sense of some changes, where Y was replaced by X. So code-reviewers naturally need to know not how Y works in a vacuum, but how is Y different from X. In this context saying something about Y in the style "...but does that" kind of implies this is that difference

Reply:
IMHO there is nothing to discuss or object. The commit message contains the exact pointed reason of a change - the key difference in consideration of owner and admin cases explained in the most clear way- and provides more than enough literally correct information for reviewer. It does not have to explain how PT code works in every detail and it does not have to explain every other difference or similarity in PT code methods. Once again - this change was done because of that exact difference in methods, when one considers owner and admin case, and other does not. All other topics together with all assumptions about how exactly methods work are simply out of the scope of my commit and out of its context.

Reply:
I agree, I should not have said `but` ending.